### DISCUSSION TITLE
inherited collections or inherited tables

### DISCUSSION THREAD

**Discussion by mkduew** [directus/directus#22794] [Feature Requests]
### Summary

a default feature in a PIM is missing in Directus

### Motivation

To make Directus a real complete PIM

### Detailed Design

In a Product Information Management (PIM) system, inheritable collections or tables play a crucial role in optimizing data organization and management. This feature allows for the management of common attributes or data sets in a hierarchical structure, where subordinate entities (such as products or categories) can inherit properties from their parent entities. Here’s a more detailed description of the functionality of inheritable collections or tables in a PIM:

1. Definition and Usage
Inheritable collections or tables are data structures that enable certain data or attributes defined at a higher level to be inherited by lower levels. This means that if an attribute or property is set at the highest level (for example, in a main category), it automatically becomes available for all subordinate levels (such as subcategories or individual products).

2. Benefits
- Efficiency Improvement: Reduces the need to repeatedly enter the same information in multiple areas of the system.
- Consistency: Ensures that all relevant objects share the same basic information, enhancing data integrity.
- Flexibility: Users can make specific adjustments or exceptions at subordinate levels without compromising the overall structure.
- Time Saving: Significantly speeds up the process of product updates and maintenance.

3. Application Examples
- Product Variants: Basic information like brand name, manufacturer, and material can be defined at the product line level and inherited by all variants of that line.
- Categorization: Attributes defined in a product category such as "Organic" or "Sustainably made" can automatically apply to all products within that category.
- Pricing: Base pricing strategies or currency settings can be set at the top level and adopted for all products within a specific group or category.

4. Technical Implementation
Technically, inheritance is often implemented using references within database tables, where "Foreign Keys" or similar mechanisms are used to define and manage relationships between levels. Changes at higher levels are propagated to subordinate levels through cascading operations.

5. Challenges
- Avoiding Data Overload: Careful planning is required to decide which data should be inherited to prevent overburdening subordinate levels with irrelevant data.
- Managing Exceptions: Handling exceptions, where subordinate elements have divergent properties, can be complex.
- Performance Optimization: Large volumes of data and complex inheritance hierarchies can impact system performance.

Implementing inheritable collections and tables in a PIM provides a robust method to simplify and accelerate product data maintenance, but it requires careful planning and management to achieve the best results.

**Comment by fitu86** [comment]
Las categorías sirven para agrupar cosas que comparten el mismo valor de alguna de sus propiedades, podemos decir que color es un atributo de las cosas, si agrupamos las cosas por el atributo color tendríamos por ejemplo categorías de cosas verdes, azules y amarillas.
